reddit_subreddit_block.py

```diff
@@ -1,16 +1,157 @@
-from nio.common.block.base import Block
+from .http_blocks.rest.rest_block import RESTPolling
 from nio.common.discovery import Discoverable, DiscoverableType
 from nio.metadata.properties import VersionProperty
+from nio.metadata.properties import PropertyHolder, StringProperty, \
+    ObjectProperty, BoolProperty
+from nio.common.signal.base import Signal
 
+import requests
+
+class Creds(PropertyHolder):
+    """ PropertyHolder for Reddit credentials. """
+    client_id = StringProperty(title='Client ID', default='[[REDDIT_CLIENT_ID]]')
+    app_secret = StringProperty(title='App Secret', default='[[REDDIT_SECRET]]')
+    app_username = StringProperty(title='App Username', default='[[REDDIT_USERNAME]]')
+    app_password = StringProperty(title='App Password', default='[[REDDIT_PASSWORD]]')
+
+class SubredditSignal(Signal):
+    """ Translates attributes from subreddit posts into Signal format """
+    def __init__(self, data):
```

Comment:
This constructor already exists on the base signal class, we should be good to just do this:
```python
class SubredditSignal(Signal):
    pass
```